src/wfcomman.c

```diff
@@ -550,8 +550,11 @@ CreateDirWindow(
 
 	   if (hwndT = HasDirWindow(hwndActive))
 	   {
+		   WCHAR szFileSpec[MAXPATHLEN];
+
 		   AddBackslash(szPath);                   // default to all files
-		   SendMessage(hwndT, FS_GETFILESPEC, MAXFILENAMELEN, (LPARAM)(szPath + lstrlen(szPath)));
+		   SendMessage(hwndT, FS_GETFILESPEC, COUNTOF(szFileSpec), (LPARAM)szFileSpec);
+		   lstrcat(szPath, szFileSpec);

```

Comment:
I don't understand how this is safe.  In treectl.c it works because szPath is MAXPATHLEN * N; here it's MAXPATHLEN + 2 but this is appending a second MAXPATHLEN to it.

Reply:
CreateDirWindow makes an implicit totally undocumented assumption, that the first parameter szPath must be MAXFILENAMELEN * 2 so that things do no crash here. 

OpenOrEditSelection() ran into that problem, and allocated less space. ( and it crashed ). 

We could solve this in two ways:
* Search for all innvocations of CreateDirWindow() and see if szPath is MAXPATHLEN * 2. Maybe we find places where where this paramete is passed into the caller than our problem spreads, and we have totally unmotivated allocations for MAXPATHLEN * 2 all over the source ( We do have this pattern in other calls already)
* We try to keep the problem local to CreateDirWindow() and solve it there. That's my approach.

Reply:
Does this part of the change fix any crash? As far as I can tell, if the buffer were too small the previous code would crash in SendMessage, this change moves that to strcat, but it will walk off the end of the same buffer in the exact same way.